monai/transforms/post/array.py

```diff
@@ -179,6 +187,11 @@ def __call__(
         if threshold_values or self.threshold_values:
             img = img >= (self.logit_thresh if logit_thresh is None else logit_thresh)
 
+        rounding = self.rounding if rounding is None else rounding
+        if rounding is not None:
+            rounding = look_up_option(rounding, ["torchrounding"])
+            img = torch.round(img)
```

Comment:
@Nic-Ma It might be worth returning this here as integer

```python
return torch.round(img).int()
```
After rounding this can only be integer so returning it as float in the next line does not make sense.

Reply:
Hi @Spenhouet ,

Thanks for your review.
Here we always return `float` because the following logic(metrics, etc.) usually expect float input and we faced several issues about `Long` tensor in computation with float tensor.
I think we can keep the existing behavior to be non-breaking, if you really expect `int`, you can add `CastToType` transform after it.
And of course, if we receive more feedback from users that ask for `int` return type, we can enhance it later.

Thanks.